org.opent2t.sample.binaryswitch.superpopular/com.smartthings.binaryswitch/js/thingTranslator.js

```diff
@@ -3,10 +3,19 @@
 var OpenT2TError = require('opent2t').OpenT2TError;
 var OpenT2TConstants = require('opent2t').OpenT2TConstants;
 var OpenT2TLogger = require('opent2t').Logger;
+var crypto = require('crypto');
 
 // This code uses ES2015 syntax that requires at least Node.js v4.
 // For Node.js ES2015 support details, reference http://node.green/
 
+/**
+ * Generate a GUID for given an ID.
+ */
+function generateGUID(stringID) {
```

Comment:
Yes. I think there is a discussion about moving a couple common functions like "validateArgumentType" and "findResource" to a library. However, since I have not heard an update on it, I put it as separate functions in each translator like other common functions.

---
In reply to: [107280991](https://github.com/openT2T/translators/pull/217#discussion_r107280991) [](ancestors = 107280991)